rama-net/src/client/pool.rs

```diff
@@ -585,13 +523,110 @@ impl<Storage, R> PooledConnectorLayer<Storage, R> {
     generate_field_setters!(wait_for_pool_timeout, Duration);
 }
 
-impl<S, Storage, R: Clone> Layer<S> for PooledConnectorLayer<Storage, R> {
-    type Service = PooledConnector<S, Storage, R>;
+impl<S, P: Clone, R: Clone> Layer<S> for PooledConnectorLayer<P, R> {
+    type Service = PooledConnector<S, P, R>;
 
     fn layer(&self, inner: S) -> Self::Service {
         PooledConnector::new(inner, self.pool.clone(), self.req_to_conn_id.clone())
             .maybe_with_wait_for_pool_timeout(self.wait_for_pool_timeout)
     }
+
+    fn into_layer(self, inner: S) -> Self::Service {
+        PooledConnector::new(inner, self.pool, self.req_to_conn_id)
+            .maybe_with_wait_for_pool_timeout(self.wait_for_pool_timeout)
+    }
+}
+
+#[cfg(feature = "http")]
+pub mod http {
+    use std::time::Duration;
+
+    use super::{FiFoReuseLruDropPool, PooledConnector, ReqToConnID};
+    use crate::{Protocol, address::Authority, client::pool::OpaqueError, http::RequestContext};
+    use rama_core::Context;
+    use rama_http_types::Request;
+
+    #[derive(Debug, Default)]
```

Comment:
shouldn't these be `Clone` as well otherwise `PoolConnector` etc won't be clone-able? And anyway doesn't hurt for these to be clone either way